demo/actions.py

```diff
@@ -792,3 +793,51 @@ def run(self, dispatcher, tracker, domain):
             )
 
         return []
+
+
+def tag_convo(tracker, label):
+    endpoint = f"http://{config.rasa_x_host}/api/conversations/{tracker.sender_id}/tags"
+    requests.post(url=endpoint, data=label)
+
+
+class ActionTagFeedback(Action):
+    """Tag a conversation in Rasa X as positive or negative feedback """
+
+    def name(self):
+        return "action_tag_feedback"
+
+    def run(self, dispatcher, tracker, domain) -> List[EventType]:
+
+        feedback = tracker.get_slot("feedback_value")
+
+        if feedback == "positive":
+            label = '[{"value":"postive feedback","color":"76af3d"}]'
+        elif feedback == "negative":
+            label = '[{"value":"negative feedback","color":"ff0000"}]'
+        else:
+            return []
+
+        tag_convo(tracker, label)
+
+        return []
+
+
+class ActionTagDocsSearch(Action):
+    """Tag a conversation in Rasa X according to whether the docs search was helpful"""
+
+    def name(self):
+        return "action_tag_docs_search"
+
+    def run(self, dispatcher, tracker, domain) -> List[EventType]:
```

Comment:
are the other action's `run()` methods typed? if so, let's add types, if not, don't worry about it

Reply:
the method returns are typed, but not the arguments